File: java/algorithm/src/test/java/org/apidesign/demo/sieve/eratosthenes/PrimesTest.java
package org.apidesign.demo.sieve.eratosthenes;

import static org.testng.AssertJUnit.assertEquals;
import org.testng.annotations.Test;

public class PrimesTest {
    @Test
    public void fifthThousandThPrime() {
        Primes p = new Primes() {
            @Override
            protected void log(String msg) {
            }
        };
        int last = p.compute();
        assertEquals("100000th prime is", 1_299_709, last);
    }
}

File: java/algorithm/src/test/java/org/apidesign/demo/sieve/eratosthenes/PrimesTest.kt
package org.apidesign.demo.sieve.eratosthenes;

import org.testng.AssertJUnit;
import org.testng.annotations.Test;

public class PrimesTest {
    class PrimesNoLog : Primes() {
        override fun log(msg: String) {}
    }

    @Test
    public fun fifthThousandThPrime() {
        val p : Primes = PrimesNoLog()
        var last : Int = p.compute();
        AssertJUnit.assertEquals("100000th prime is", 1_299_709, last);
    }
}
